src/main/java/org/spongepowered/common/mixin/core/world/MixinWorldProvider.java

```diff
@@ -115,29 +94,14 @@ public boolean hasSky() {
         return !getHasNoSky();
     }
 
-    @Overwrite
-    public boolean getHasNoSky() {
-        return this.terrainType.equals(GeneratorTypes.NETHER) || hasNoSky;
-    }
-
-    public String mixinworldprovider$getSaveFolder() {
-        final IMixinWorldProvider this$ = (IMixinWorldProvider) this;
-        return (this$.getDimensionId() == 0 ? null : "DIM" + this$.getDimensionId());
-    }
-
-    public void mixinworldprovider$setDimensionConfig(SpongeConfig<SpongeConfig.DimensionConfig> config) {
-        this.dimensionConfig = config;
-    }
-
-    public SpongeConfig<SpongeConfig.DimensionConfig> mixinworldprovider$getDimensionConfig() {
-        return this.dimensionConfig;
+    @Override
+    public void setGeneratorSettings(String generatorSettings) {
+        this.generatorSettings = generatorSettings;
     }
 
+    // TODO 1.9 - Make DimensionType a ContextSource and not dimension? -- Zidane
```

Comment:
TODO for myself. DimensionType should be the context source. Need to change this.